runtime/src/main/java/org/corfudb/protocols/logprotocol/SMREntryGarbageInfo.java

```diff
@@ -0,0 +1,110 @@
+package org.corfudb.protocols.logprotocol;
+
+
+import io.netty.buffer.ByteBuf;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+import org.corfudb.runtime.CorfuRuntime;
+
+import javax.naming.OperationNotSupportedException;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * SMREntryGarbageInfo maintains garbage information of one SMREntry. This SMREntry could be stand-alone or a part of
+ * MultiSMREntry. The existence of SMREntryGarbageInfo itself indicates the associated SMREntry has been identified
+ * as garbage. SMREntryGarbageInfo could be used individually or composites more complicated ISMRGarbageInfo.
+ */
+@ToString(callSuper = true)
+@NoArgsConstructor
+public class SMREntryGarbageInfo extends LogEntry implements ISMRGarbageInfo {
+
+    /**
+     * Global address of the SMREntry that supersedes the associated SMREntry.
+     */
+    @Getter
+    private long detectionAddress;
+
+    /**
+     * The size of the associated SMREntry in Byte.
+     */
+    private transient int smrEntrySize;
+
+    /**
+     * Constructor for SMREntryGarbageInfo
+     * @param detectionAddress
+     * @param smrEntrySize
+     */
+    public SMREntryGarbageInfo(long detectionAddress, int smrEntrySize) {
+        super(LogEntryType.SMR_GARBAGE);
+        this.detectionAddress = detectionAddress;
+        this.smrEntrySize = smrEntrySize;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Optional<SMREntryGarbageInfo> getGarbageInfo(UUID streamId, int index) {
+        // TODO(Xin): we should check that the id matches the id of this entry
+        return Optional.of(this);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Map<Integer, SMREntryGarbageInfo> getAllGarbageInfo(UUID streamId) {
+        Map<Integer, SMREntryGarbageInfo> garbageInfo = new ConcurrentHashMap<>();
+        garbageInfo.put(0, this);
+        return garbageInfo;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getGarbageSize() {
+        return smrEntrySize;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void remove(UUID streamId, int index) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void merge(ISMRGarbageInfo info) {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void add(UUID streamId, int index, SMREntryGarbageInfo smrEntryGarbageInfo) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void serialize(ByteBuf b) {
```

Comment:
Serialize the size field as we discussed.

Reply:
done.